qa-tests-backend/src/test/groovy/ProcessBaselinesTest.groovy

```diff
@@ -218,24 +218,33 @@ class ProcessBaselinesTest extends BaseSpecification {
         assert deploymentId != null
 
         String containerName = deployment.getName()
-        ProcessBaselineOuterClass.ProcessBaseline baseline = ProcessBaselineService.
+        // Need to make sure the processes show up before we lock.
+        def baseline = evaluateWithRetry(30, 4) {
+            def tmpBaseline = ProcessBaselineService.
                  getProcessBaseline(clusterId, deployment, containerName)
+            if (tmpBaseline.elementsList.size() == 0) {
+                throw new RuntimeException(
+                    "No processes in baseline for deployment ${deploymentId} yet. Baseline is ${tmpBaseline}"
+                )
+            }
+            return tmpBaseline
+        }
+
         assert (baseline != null)
+        log.info "Baseline Before locking: ${baseline}"
         assert ((baseline.key.deploymentId.equalsIgnoreCase(deploymentId)) &&
                  (baseline.key.containerName.equalsIgnoreCase(containerName)))
         assert baseline.elementsList.find { it.element.processName == processName } != null
 
         List<ProcessBaselineOuterClass.ProcessBaseline> lockProcessBaselines = ProcessBaselineService.
                  lockProcessBaselines(clusterId, deployment, containerName, true)
         assert (!StringUtils.isEmpty(lockProcessBaselines.get(0).getElements(0).getElement().processName))
-        // sleep 30 seconds to allow for propagation to sensor
-        sleep 30000
-        orchestrator.execInContainer(deployment, "pwd")
 
-        log.info "Locked Process Baseline: ${lockProcessBaselines}"
+        // sleep 5 seconds to allow for propagation to sensor
+        sleep 5000
+        orchestrator.execInContainer(deployment, "pwd")
 
-        // Give the reprocessing some time.
-        sleep 30000
```

Comment:
It looks like we removed two sleeps here. Maybe in face of flakes, it would be better to change one thing at a time? or are you confident about these removals?

Reply:
well I did do it one at a time just within the PR.  Once multiple runs were successful after the first one, I removed the second and had multiple successful runs.  Now of course that doesn't mean once it is merged that we won't find some nightly environment where it fails.  In that event I would just recommend we write a ticket and ignore the test while we are working on it.